### ISSUE TITLE
Webstorm doesn't recognize step definitions

### ISSUE THREAD

**Issue by kayvanbree** [TheBrainFamily/cypress-cucumber-preprocessor#56]
I don't really know if I should ask this here, but Webstorm doesn't recoginize the step definitions of this plugin because the steps are given, when, then. Webstorm does recognize the official cucumber package's Given, When an Then (capitalized).  Webstorm recognizing step definitions is extremely helpful in preventing duplicate step definitions. 

```
// Recognized (like the official cucumber package)
Given(`I am on the webpage`, () => {
  cy.visit(url);
});

// Not recognized (like this plugin)
given('I am on the webpage', () => {
  cy.visit(url);
});
```
![webstormnotrecognizingcypresscucumber](https://user-images.githubusercontent.com/11921540/41404040-7e0da9be-6fc6-11e8-9c2a-2e9ebeaba719.gif)

I did find a workaround, but I'd rather not have people forget to include this:

```
// Just add this above the step definitions
const Given = given;
const When = when;
const Then = then;
```

Is there a way this could be solved without a workaround?

**Comment by lgandecki** [comment]
I was thinking about this as well... 
I don't really like the fact that the cucumber functions are called with capitalized letter (suggesting a class),
but if the existing tooling relies on that - let's add it. For the backward compability we would have to export given/when/then as both Capitalized and lowerCase.

@kayvanbree Would you like to send a PR for this?

Adding a simple test - where you define the steps with Capitalized letter would be appreciated as well.

**Comment by kayvanbree** [comment]
@lgandecki I don't like Cucumbers way either, but I think they maybe did it to have capitalized letters in Gherkin files. Anyway, I'll send over a PR when I have time!

**Comment by kayvanbree** [comment]
I got Webstorm to recognize the steps, but now only ScenarioOutline.feature passes. The rest of the tests fail:

```
Oops...we found an error preparing this test file:

  cypress\integration\DataTable.feature

The error was:

Error: Cannot find module '..supportstep_definitions/requiringFilesData' from 'C:\Projects\cypress-cucumber-preprocessor\cypress\integration'

This occurred while Cypress was compiling and bundling your test code. This is usually caused by:

- A missing file or dependency
- A syntax error in the file or one of its dependencies
```

I'll have to get back to this later.

**Comment by lgandecki** [comment]
very weird, the changes you made shouldn't affect anything. Can you open a PR here, circle should catch it and run tests on it - maybe the code is correct but it's some weirdness in your local setup.

**Comment by kayvanbree** [comment]
Created #64, let's see what Circle thinks

**Comment by kayvanbree** [comment]
I can't get this error on my machine... I still get the parsing error with the incorrect directory.

**Comment by kayvanbree** [comment]
I don''t really get how stuff works, but I see there's a .toLowerCase in resolveStepDefinition.js.

```
function resolveStepDefinition(step) {
  const stepDefinition = stepDefinitionRegistry.resolve(
    step.keyword.toLowerCase().trim(),
    step.text
  );

  return stepDefinition || {};
}
```

**Comment by lgandecki** [comment]
the reason for different errors might be some difference between windows and unix machine. I will try to fix this up today (having your PR helps). It shouldn't be too hard now.
Thanks for your help :-) And will update you here in a few hours

**Comment by lgandecki** [comment]
I got this to work, but your PR is weird. Please DO NOT change anything in package.json and don't commit package-lock.json.
you've just added the exports, but you don't import them anywhere. Actually, there is no need to export them, the only place that you have to change is index.js, by adding:

```javascript

  const Given = given;
  const When = when;
  const Then = then;

```
here:

<img width="1439" alt="screen shot 2018-07-04 at 20 53 06" src="https://user-images.githubusercontent.com/4002543/42288576-5c28d952-7fcc-11e8-8c33-26ffe1e2fe92.png">

we could export only given and alias everything to given anyway, but I left the duplicated given/when/then exports for backward compability. 

Interesting thing though, that didn't make it work for me in webstorm.. when I tried this initially, it didn't work neither (it also did not work for a normal cucumber 2+ project)

Is there a chance you could push somwhere the code that you were showcasing this problem in the first post? I'd take a look if this is about a folder structure, or what's different..

**Comment by vadimromanyak** [comment]
@lgandecki Webstorm doesn't recognize step definition this way:
Given('user navigated to the Start page', () => { });

But if we replace quotes with /^...?/ then it works:
Given(/^user navigated to the Start page?/, () => { });

**Comment by kayvanbree** [comment]
@lgandecki 

I will change my PR and check if the build works when I do the `const Given = given` stuff. 

@vadimromanyak You can also use ticks:

``` 
Given(\`user navigated to the Start page\`, () => {});
```

If you check out the test steps I created you can see I use either the ticks or /^..?/:

```
When(/^I write my steps using caps$/, () => {
  counter += 1;
});
```

**Comment by lgandecki** [comment]
@kayvanbree Bingo! backticks work!

**Comment by lgandecki** [comment]
@vadimromanyak replacing with /^ etc is not preferable as then we lose the prefered / new / cleaner syntax:

https://github.com/cucumber/cucumber-js/blob/master/docs/nodejs_example.md

But I'm excited that the backticks work, not sure why webstorm wouldn't work with " or ' but hey, not a huge deal once you know it :)

**Comment by kayvanbree** [comment]
@lgandecki My local tests are failing again. I'll commit to see what Circle thinks. 

Also, the /^ stuff may not be preferable, but it's the only way I got `(\d+)` and (.*?) working to use numbers and strings in my step definitions.

**Comment by lgandecki** [comment]
@kayvanbree well, yeah, the /^ gives you the regexp functionality.. but the point is, instead of using regexp, you can just use the newer syntax... 

to give you an example, instead of:

```javascript
then(/(.*?) is chosen/, choice => {})
then(/my lucky number is (\d+)/, number => {})
```

```javascript
then(`{word} is chosen`, choice => {})
then(`my lucky number is {int}`, number => {})
```

Regexps are powerful, but unreadable to human, and very rarely u need such a complex processing anyway, usually it's a word, or a number, etc.. so the cucumber guys introduced the new syntax :)

**Comment by kayvanbree** [comment]
@lgandecki The tests for #64 are passing!

**Comment by kayvanbree** [comment]
I don't think webstorm plays nice with the new syntax...

**Comment by kayvanbree** [comment]
Indeed

**Comment by lgandecki** [comment]
closed in #65 , it works with us as much as it works with pure cucumber, not much more we can't do at this time.
Thanks @kayvanbree @vadimromanyak for your help here :)

**Comment by san-ouadghiri** [comment]
https://youtrack.jetbrains.com/issue/WEB-32819 for follow-up. You can "thumb-up" to make it gain interest, and/or watch the issues for updates.
